packages/common/src/services/coingecko/CoingeckoClient.ts

```diff
@@ -0,0 +1,65 @@
+import { RateLimiter } from '../../tools/RateLimiter'
+import { HttpClient } from '../HttpClient'
+import { CoingeckoId, MarketChartRangeParams } from '.'
+import {
+  CoinListEntry,
+  CoinListPlatformEntry,
+  CoinListPlatformResult,
+  CoinListResult,
+  CoinMarketChartRangeResult,
+} from './model'
+
+const API_URL = 'https://api.coingecko.com/api/v3'
+
+export class CoingeckoClient {
+  private rateLimiter = new RateLimiter({
+    callsPerMinute: 40,
+  })
+  private timeoutMs = 10_000
+
+  constructor(private httpClient: HttpClient) {
+    this.query = this.rateLimiter.wrap(this.query.bind(this))
+  }
+
+  async getCoinList(options?: {
+    includePlatform: false
+  }): Promise<CoinListEntry[]>
+  async getCoinList(options: {
+    includePlatform: true
+  }): Promise<CoinListPlatformEntry[]>
+  async getCoinList(options = { includePlatform: false }) {
+    const data = await this.query('/coins/list', {
+      include_platform: options.includePlatform.toString(),
+    })
+    if (!options.includePlatform) {
+      return CoinListResult.parse(data)
+    } else {
+      return CoinListPlatformResult.parse(data)
+    }
+  }
+
+  async getCoinMarketChartRange(
+    coindId: CoingeckoId,
+    params: MarketChartRangeParams
```

Comment:
I would not have an object here. Rather I'd take multiple parameters and construct the object in the function.

Reply:
Consider using the existing UnixTimestamp class